Content.Client/Xenoarchaeology/Ui/AnalysisConsoleMenu.xaml.cs

```diff
@@ -15,169 +24,226 @@ namespace Content.Client.Xenoarchaeology.Ui;
+        };
+
+        ExtractButton.OnPressed += StartExtract;
 
-        var buttonGroup = new ButtonGroup(false);
-        UpBiasButton.Group = buttonGroup;
-        DownBiasButton.Group = buttonGroup;
+        var comp = _ent.GetComponent<AnalysisConsoleComponent>(owner);
+        _owner = (owner, comp);
+        Update((owner, comp));
+        SetSelectedNode(null);
     }
 
-    protected override void FrameUpdate(FrameEventArgs args)
+    private void StartExtract(BaseButton.ButtonEventArgs obj)
     {
-        base.FrameUpdate(args);
-
-        if (_startTime is not { } start || _totalTime is not { } total || _accumulatedRunTime is not { } accumulated)
+        if (!_artifactAnalyzer.TryGetArtifactFromConsole(_owner, out var artifact))
             return;
 
-        var remaining = total - accumulated;
-        if (!_paused)
+        ExtractContainer.Visible = true;
+        NodeViewContainer.Visible = false;
+
+        _nodeExtractionValues.Clear();
+        _extractionSum = 0;
+        _extractionMessage.Clear();
+        _nextExtractStringTime = _timing.CurTime;
+
+        var nodes = _xenoArtifact.GetAllNodes(artifact.Value);
+        foreach (var node in nodes)
         {
-            // If the analyzer is running, its remaining time is further discounted by the time it's been running for.
-            remaining += start - _timing.CurTime;
+            var pointValue = _xenoArtifact.GetResearchValue(node);
+            if (pointValue <= 0)
+                continue;
+            var nodeId = _xenoArtifact.GetNodeId(node);
+
+            var text = Loc.GetString("analysis-console-extract-value", ("id", nodeId), ("value", pointValue));
+            _nodeExtractionValues.Add((text, pointValue));
         }
-        var secsText = Math.Max((int) remaining.TotalSeconds, 0);
 
-        ProgressLabel.Text = Loc.GetString("analysis-console-progress-text",
-            ("seconds", secsText));
+        if (_nodeExtractionValues.Count == 0)
+            _nodeExtractionValues.Add((Loc.GetString("analysis-console-extract-none"), 0));
 
-        // 1.0 - div because we want it to tick up not down
-        ProgressBar.Value = Math.Clamp(1.0f - (float) remaining.Divide(total), 0.0f, 1.0f);
+        _nodeExtractionValues.Sort((x, y) => x.Item2.CompareTo(y.Item2));
     }
 
-    public void SetButtonsDisabled(AnalysisConsoleUpdateState state)
+    protected override void FrameUpdate(FrameEventArgs args)
     {
-        ScanButton.Disabled = !state.CanScan;
-        PrintButton.Disabled = !state.CanPrint;
-        if (state.IsTraversalDown)
-            DownBiasButton.Pressed = true;
-        else
-            UpBiasButton.Pressed = true;
+        base.FrameUpdate(args);
+
+        if (_nextExtractStringTime == null)
+            return;
+
+        if (_timing.CurTime < _nextExtractStringTime)
+            return;
 
-        ExtractButton.Disabled = false;
-        if (!state.ServerConnected)
+        if (_nodeExtractionValues.Count == 0)
```

Comment:
this is some really evil shit. it basically controls left panel that contains both 'current node info' data AND 'extract points' data at the same time. It might be better to make separate control, but its SO specific... it makes some nice stuff like 'hide extract points info faster if no useful info is shown' but man... this code in FrameUpdate IS scary and having it here is error prone IMO

Reply:
this is 'node viewer' version
![image](https://github.com/user-attachments/assets/4e49efe0-151d-4dbe-9f89-26c5e64dbd7d)

Reply:
this is panel with extract. it closes pretty quickly. 
![image](https://github.com/user-attachments/assets/2b162ccb-ad86-45d9-88a0-9fa79ae61094)